README.md

```diff
@@ -34,608 +36,95 @@ For more information refer to [IntelliJ IDEA documentation](https://www.jetbrain
+container both from within a container and from the host system.
 
-`-e wall` option tells async-profiler to sample all threads equally every given
-period of time regardless of thread status: Running, Sleeping or Blocked.
-For instance, this can be helpful when profiling application start-up time.
+For more details, please refer to
+[Profiling In Container Documentation](https://github.com/async-profiler/async-profiler/blob/master/docs/ConverterUsage.md).
 
-Wall-clock profiler is most useful in per-thread mode: `-t`.
+# Profiling Non-Java applications
 
-Example: `asprof -e wall -t -i 5ms -f result.html 8983`
+Unlike traditional Java profilers, async-profiler monitors non-Java threads (e.g., GC threads)
+and also shows native frames in Java stack traces. This enables it to work with C and C++
+applications.
 
-## Java method profiling
+For more details, please refer to
+[Profiling Non-Java Applications](https://github.com/async-profiler/async-profiler/blob/master/docs/ProfilingNonJavaApplications.md).
 
-`-e ClassName.methodName` option instruments the given Java method
-in order to record all invocations of this method with the stack traces.
+# JFR Visualization
 
-Example: `-e java.util.Properties.getProperty` will profile all places
-where `getProperty` method is called from.
+`jfr` output from async-profiler can be visualized in a human-readable format using multiple
+options, listed and explained in the
+[JFR Visualization Documentation](https://github.com/async-profiler/async-profiler/blob/master/docs/JfrVisualization.md).
 
-Only non-native Java methods are supported. To profile a native method,
-use hardware breakpoint event instead, e.g. `-e Java_java_lang_Throwable_fillInStackTrace`
-
-**Be aware** that if you attach async-profiler at runtime, the first instrumentation
-of a non-native Java method may cause the [deoptimization](https://github.com/openjdk/jdk/blob/bf2e9ee9d321ed289466b2410f12ad10504d01a2/src/hotspot/share/prims/jvmtiRedefineClasses.cpp#L4092-L4096)
-of all compiled methods. The subsequent instrumentation flushes only the _dependent code_.
-
-The massive CodeCache flush doesn't occur if attaching async-profiler as an agent.
-
-Here are some useful native methods that you may want to profile:
-* ```G1CollectedHeap::humongous_obj_allocate``` - trace _humongous allocations_ of the G1 GC,
-* ```JVM_StartThread``` - trace creation of new Java threads,
-* ```Java_java_lang_ClassLoader_defineClass1``` - trace class loading.
-
-## Building
+# Building
 
 Build status: [![Build Status](https://github.com/async-profiler/async-profiler/actions/workflows/ci.yml/badge.svg?branch=master)](https://github.com/async-profiler/async-profiler/actions/workflows/ci.yml)
 
+## Build requirements
+* JDK: 11+
+* GCC: 7.5.0+
+
 Make sure the `JAVA_HOME` environment variable points to your JDK installation,
 and then run `make`. GCC or Clang is required. After building, the profiler binaries
 will be in the `build` subdirectory.
```

Comment:
This deserves a separate chapter with a description of Makefile targets and parameters and instructions how to produce compatible binaries. May be in a separate PR.

Reply:
Removed now as suggested